parser/lexer/state.go

```diff
@@ -28,6 +34,8 @@ func root(l *lexer) stateFn {
 		l.emit(Bracket)
 	case strings.ContainsRune(")]}", r):
 		l.emit(Bracket)
+	case strings.ContainsRune("&", r) && !strings.ContainsRune("&", rNext):
```

Comment:
I'm not sure it's necessary since "*" versus "**" doesn't use such a condition. Could you try out going trough the " possible double rune operator" condition ?

Reply:
done :)

Reply:
I finally kept this pattern to be able to use constant variables for easier eventual syntax changes. Do you agree?

Reply:
I'm not sure I do agree, this is introducing a new pattern of code in the library and I'm really not sure a syntax change is a possibility since it would be a big issue for the library and mostly for persisted objects in production. @Pikipoo would you comment on this ?

Reply:
(constant could still be used and if you wan't to introduce a good syntax do it at compiled time or init time, not run time) -> you can compute single and double chains from a complete op const list for instance.
I actually like the original lexer pattern, it seems optimum here.

Reply:
I don't understand your point here. How to use those constants then?